### PR TITLE
Adding more on-prem install instructions

### PR DESCRIPTION
**Pull request by arschles**
TODO
- [x] consolidate system requirements into a new section (right now it's in intro and install)
- [x] organize setup by the [single_server, HA] then by [Docker/manual, Terraform] - moving Terraform and CoreOS into separate sections instead
- [ ] add a FAQ page
- [ ] clean up `mq/3/on-premise/management`
- [x] more on load balancing
- [x] centralize system requirements

### CODE REVIEW

**Comment by arschles** [comment]
@Stephenitis PTAL

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM razic/bundler
+
+RUN mkdir -p /iron/docs
+ADD Gemfile /iron/docs/Gemfile
+WORKDIR /iron/docs
+RUN bundle install
+
+ENTRYPOINT ["bundle"]
```
Comment (arschles) [inline Dockerfile:8]: this doesn't work yet. will include a `dockerserve.sh` to serve more easily

**Comment by arschles** [comment]
@Stephenitis @carimura @treeder lmk what you think. main install guide is there there, moved TF and CoreOS install to the side

**Comment by treeder** [comment]
LGTM, let's merge and iterate if no errors.

**Pull request merged**